cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -599,3 +603,46 @@ def test_guest_os_shutdown(appliance, provider, testing_vm_tools, ensure_vm_runn
         new_last_boot_time = view.entities.summary("Power Management").get_text_of("Last Boot Time")
         soft_assert(new_last_boot_time == last_boot_time,
                     "ui: {} should ==  orig: {}".format(new_last_boot_time, last_boot_time))
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1687597])
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE, override=True)
+def test_retire_vm_with_vm_user_role(request, appliance, testing_vm):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        startsin: 5.10
+        casecomponent: Automate
+
+    Bugzilla:
+        1687597
+    """
+    # Using group - "EvmGroup-vm_user" because this is attached to role - "EvmRole-vm_user"
+    user_group = appliance.collections.groups.instantiate(description="EvmGroup-vm_user")
+
+    # Creating a user with 'EvmGroup-vm_user' group to retire VM provisioned by admin
+    new_user = appliance.collections.users.create(
```

Comment:
How about using REST to create the role, group, and user? That you will only need to create and instantiate them, no need to add finalizer to delete them.
If it doesn't affect your test, I think using REST will make your test case much faster.

Reply:
I think the role, user, and group should be moved into fixtures (possibly only one fixture), since they are out of the scope of the test. And I definitely agree with using REST -- we should use rest anywhere possible to speed up our test cases. Thoughts?